### PR TITLE
docs(config options): `automerge` and GitHub branch protection rule

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Add warning admonition, if you use GitHub's **Require status checks to pass before merging** branch protection rule, then you must select at least one status check in that section

## Context

Users who select the **Require status checks to pass before merging** checkbox expect GitHub to block Renovate PRs with any failing check. But for GitHub's feature to work properly, you must actually select _at least one status check_.

Examples of users confused by this behavior:

- #25750
- https://github.com/orgs/community/discussions/63427

I don't know when/if GitHub will improve the form's text and layout. In the meantime we can warn our own users about this behavior.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
This docs addition is unintentionally misleading.

Renovate never merges a PR with failing tests itself. The sentence implies that Renovate is a bandit which will merge your failing PR if you don't forbid it, and this is not the case.

Instead, this applies to platform automerge. If platform automerge is used, Renovate tells GitHub "please automerge this when it's ready". It is GitHub which decides "who cares about the failing tests" and merges, not Renovate

**Comment by HonkingGoose** [comment]
> This docs addition is unintentionally misleading.

Yeah it's tricky to get the right wording for this. I don't want to blame GitHub (too much), and avoid users thinking Renovate is doing something wrong... 😄 

> If you don't GitHub allows Renovate to automerge PPs with failing tests!

I expected this wording to be about right. 🙃 

> Renovate never merges a PR with failing tests itself. The sentence implies that Renovate is a bandit which will merge your failing PR if you don't forbid it, and this is not the case.

🤠 Yeehaw, pardner. Allow me to automerge this here PR. I know it has breaking tests, but around these parts we like to live dangerously.

> Instead, this applies to platform automerge. If platform automerge is used, Renovate tells GitHub "please automerge this when it's ready". It is GitHub which decides "who cares about the failing tests" and merges, not Renovate

I think Renovate defaults to `platformAutomerge=true` though, so the first thing a user is likely to do is to turn on `automerge=true`, without reading `platformAutomerge` docs.

Open for suggestions! 😄

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -199,6 +199,11 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
 !!! warning "Negative reviews on GitHub block Renovate automerge"
     Renovate won't automerge on GitHub if a PR has a negative review.
 
+<!-- prettier-ignore -->
+!!! warning "Renovate automerge and GitHub branch protection rule: Require status checks to pass before merging branch protection rule"
+    If you set `automerge: true` _and_ use GitHub's **Require status checks to pass before merging** branch protection rule, then you must select at least one status check in that section.
+    If you don't GitHub allows Renovate to automerge PPs with failing tests!
```
Comment (oliversalzburg) [inline docs/usage/configuration-options.md:205]:
```suggestion
    If you don't GitHub allows Renovate to automerge PRs with failing tests!
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -199,6 +199,11 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
 !!! warning "Negative reviews on GitHub block Renovate automerge"
     Renovate won't automerge on GitHub if a PR has a negative review.
 
+<!-- prettier-ignore -->
+!!! warning "Renovate automerge and GitHub branch protection rule: Require status checks to pass before merging branch protection rule"
+    If you set `automerge: true` _and_ use GitHub's **Require status checks to pass before merging** branch protection rule, then you must select at least one status check in that section.
+    If you don't GitHub allows Renovate to automerge PRs with failing tests!
```
Comment (rarkins) [inline docs/usage/configuration-options.md:205]:
```suggestion
    If you don't, and platform automerge is in use, then GitHub might automerge PRs with failing tests!
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -199,6 +199,11 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
 !!! warning "Negative reviews on GitHub block Renovate automerge"
     Renovate won't automerge on GitHub if a PR has a negative review.
 
+<!-- prettier-ignore -->
+!!! warning "Renovate automerge and GitHub branch protection rule: Require status checks to pass before merging branch protection rule"
+    If you set `automerge: true` _and_ use GitHub's **Require status checks to pass before merging** branch protection rule, then you must select at least one status check in that section.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:204]:
```suggestion
    If you set `automerge: true` _and_ use `platformAutomerge=true`  (Renovate defaults to true) and_ use GitHub's **Require status checks to pass before merging** branch protection rule, then you must select at least one status check in that section.
```

This is getting messy, but we should probably mention
`platformAutomerge` here.

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -205,6 +205,16 @@ So for example you could choose to automerge all (passing) `devDependencies` onl
...
 
+### Automerge and GitHub branch protection rules
+
+You must select at least one status check in the **Require status checks to pass before merging** section of your branch protection rules on GitHub, if you match all three conditions:
+
+- `automerge=true`
+- `platformAutomerge=true`, Renovate defaults to `true`
+- you use use GitHub's **Require status checks to pass before merging** branch protection rule
+
+If you don't select any status check, and you use platform automerge, then GitHub might automerge PRs with failing tests!
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:216]: I wanted to have a proper section for the advice, so I can use a list. We already have a lot of admonitions in the `automerge` docs. I don't think adding another colored box is going to help readers.

But my new `h3` section is causing a test to fail:

```
 documentation › website-documentation › configuration-options › has headers for every required sub-option

expect(received).toEqual(expected) // deep equality

- Expected  - 0
+ Received  + 1

@@ -1,6 +1,7 @@
  Array [
+   "Automerge and GitHub branch protection rules",
    "abortIgnoreStatusCodes",
    "abortOnError",
    "allowedVersions",
    "artifactAuth",
    "authType",

    at Object.<anonymous> (test/documentation.spec.ts:60:28)
```

Comment (rarkins) [reply docs/usage/configuration-options.md:216]: Unfortunately you can't use headings, as they have special meanings

**Review by rarkins — APPROVED** [review]

**Pull request merged**